Modified rails3 patterns to tolerate no indentation

marxarelli opened the pull request:
Some logging mechanisms (e.g. syslog) tend to remove indentation from
parsed log lines. The new regex patterns will cover these cases.

marxarelli committed:
Modified rails3 patterns to tolerate no indentation

Some logging mechanisms (e.g. syslog) tend to remove indentation from
parsed log lines. The new regex patterns will cover these cases.

wvanbergen commented:
Thanks for contributing. Questions:
- Do you know in what Rails version this was changed?
- I think we can also just remove the space instead of using `(?:^| )`

marxarelli commented:
The format hasn't changed in Rails—the default logger still uses 2 spaces for indentation of these lines. However, some alternative loggers such as Syslogger and SyslogLogger remove the indentation to avoid problems with syslog parsing.

Just removing the space from the regex pattern should work just fine. I just wanted to be sure there wouldn't be any unforeseen compatibility issues: A pattern without the explicit word boundary could potentially match something like "OtherParameters: ..." but the occurrence of such lines seems pretty unlikely.

wvanbergen commented:
In that case we should probably use `\b` in the regex to require a boundary.

marxarelli commented:
Using `\b` does seem more appropriate. This would also cover cases where lines are prepended by the logger (with timestamp, pid, etc.).

wvanbergen commented:
OK, if you can update your PR I'll merge it in right away. Thanks again!

marxarelli commented:
No problem. I'm glad to contribute.

marxarelli committed:
Refactored rails3 file format regex patterns

Removed BOL/space from patterns in favor of \b (word boundary) to match
lines modified by some alternative loggers.
